wremnants/datasets/datasets2016.py

```diff
@@ -8,55 +8,99 @@
 lumijson = f"{pathlib.Path(__file__).parent.parent}/data/Cert_271036-284044_13TeV_Legacy2016_Collisions16_JSON.txt"
 
 #TODO add the rest of the samples!
+def makeFilelist(paths, maxFiles=-1, eos=False):
+    filelist = []
+    for path in paths:
+        filelist.extend(glob.glob(path) if not eos else buildXrdFileList(path, "eoscms.cern.ch"))
+    return filelist if maxFiles < 0 else filelist[:maxFiles]
 
-def allDatasets(istest=False):
-    data_files = data_files_ if not istest else data_files_[:1]
-    zmc_files = zmc_files_ if not istest else zmc_files_[:1]
-    wplus_files = wplus_files_ if not istest else wplus_files_[:1]
-    wminus_files = wminus_files_ if not istest else wminus_files_[:1]
-    data = narf.Dataset(name = "dataPostVFP",
-                        filepaths = data_files,
-                        is_data = True,
-                        lumi_csv = lumicsv,
-                        lumi_json = lumijson)
-
-    zmc = narf.Dataset(name = "ZmumuPostVFP",
-                    filepaths = zmc_files,
-                    is_data = False,
-                    xsec = 1990.5,
-                    )
-
-    wplus = narf.Dataset(name = "WplusmunuPostVFP",
-                    filepaths = wplus_files,
-                    is_data = False,
-                    xsec = 1990.5,
-                    )
-
-    wminus = narf.Dataset(name = "WminusmunuPostVFP",
-                    filepaths = wminus_files,
-                    is_data = False,
-                    xsec = 1990.5,
-                    )
-
-    return [data, zmc, wplus, wminus]
-    #return [zmc, wplus, wminus]
-    #return [zmc]
-    #return [data]
-    #return [wplus, wminus]
+def getDatasets(maxFiles=-1, filt=None):
+    dataPostVFP = narf.Dataset(name = "dataPostVFP",
+        filepaths = makeFilelist(["/scratch/shared/originalNANO/NanoV8Data_Mar2021/Run2016F_postVFP/*",
+            "/scratch/shared/originalNANO/NanoV8Data_Mar2021/Run2016G/*",
+            "/scratch/shared/originalNANO/NanoV8Data_Mar2021/Run2016H/*",
+        ], maxFiles),
+        is_data = True,
+        lumi_csv = lumicsv,
+        lumi_json = lumijson
+    )
+
+    ZmmPostVFP = narf.Dataset(name = "ZmumuPostVFP",
+        filepaths = makeFilelist(
+            ["/scratch/shared/NanoAOD/DYJetsToMuMu_M-50_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/MCPostVFPWeightFix/211114_184608/000*/*.root"], maxFiles),
+        is_data = False,
+        xsec = 1976.1,
+    )
+
+    BR_TAUToMU = 0.17394
+    ZttPostVFP = narf.Dataset(name = "ZtautauPostVFP",
+        filepaths = makeFilelist(
+            ["/scratch/shared/originalNANO/DYJetsToTauTau_postVFP/*/*.root"], maxFiles),
+        is_data = False,
+        # At least one tau->e or mu decay, so everything that's not all other decays
+        xsec = ZmmPostVFP.xsec*(1.-(1.-(BR_TAUToMU+0.1782)**2)),
```

Comment:
0.1782 is BR_TAUToE?  (make it a variable like BR_TAUToMU?)

Reply:
ya ok